qiskit/quantum_info/operators/utils.py

```diff
@@ -0,0 +1,106 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Quantum information utility functions for operators.
+"""
+
+from typing import TypeVar
+
+from qiskit.quantum_info.operators.linear_op import LinearOp
+
+OperatorTypeT = TypeVar("OperatorTypeT", bound=LinearOp)
+
+
+def commutator(a: OperatorTypeT, b: OperatorTypeT) -> OperatorTypeT:
+    r"""Compute commutator of a and b.
+
+    .. math::
+
+        ab - ba.
+
+    Args:
+        a: Operator a.
+        b: Operator b.
+    Returns:
+        The commutator
+    """
+    return a @ b - b @ a
+
+
+def anti_commutator(a: OperatorTypeT, b: OperatorTypeT) -> OperatorTypeT:
+    r"""Compute anti-commutator of a and b.
+
+    .. math::
+
+        ab + ba.
+
+    Args:
+        a: Operator a.
+        b: Operator b.
+    Returns:
+        The anti-commutator
+    """
+    return a @ b + b @ a
+
+
+def double_commutator(
+    a: OperatorTypeT, b: OperatorTypeT, c: OperatorTypeT, *, commutes: bool = True
+) -> OperatorTypeT:
+    r"""Compute symmetric double commutator of a, b and c.
+    See also Equation (13.6.18) in [1].
+
+    If `commutes` is `True`, it returns
+
+    .. math::
+
+         [[A, B], C]/2 + [A, [B, C]]/2
+         = (2ABC + 2CBA - BAC - CAB - ACB - BCA)/2.
+
+    If `commutes` is `False`, it returns
+
+    .. math::
+         \lbrace[A, B], C\rbrace/2 + \lbrace A, [B, C]\rbrace/2
+         = (2ABC - 2CBA - BAC + CAB - ACB + BCA)/2.
```

Comment:
Does this render correctly in the docs or does this need another linebreak?
```suggestion
    .. math::

         \lbrace[A, B], C\rbrace/2 + \lbrace A, [B, C]\rbrace/2
         = (2ABC - 2CBA - BAC + CAB - ACB + BCA)/2.
```

Reply:
I'm looking at the rendering results, but probably not necessary.

<img width="785" alt="image" src="https://user-images.githubusercontent.com/6814928/221451296-a9f0495b-b871-4aa3-8c1b-3c6884d53c20.png">